pipeline_tools/tests/test_get_analysis_metadata.py

```diff
@@ -65,17 +68,11 @@ def _request_callback(request, context):
             return {'status': 'error', 'message': 'Internal Server Error'}
 
         mock_auth.return_value = HTTPBasicAuth('user', 'password')
-        # Makes the test complete faster by limiting the number of retries
-        get_analysis_metadata.get_metadata.retry.stop = stop_after_attempt(3)
 
         mock_request.get(self.cromwell_url, json=_request_callback)
-
-        with self.assertRaises(requests.HTTPError):
-            get_analysis_metadata.get_metadata(self.runtime_environment, self.workflow_id, use_caas=False)
-            self.assertNotEqual(mock_request.call_count, 1)
-
-        # Reset decorator default
-        get_analysis_metadata.get_metadata.retry.stop = stop_after_delay(20)
+        with self.assertRaises(requests.HTTPError), HttpRequestsManager():
+            get_analysis_metadata.get_metadata(self.runtime_environment, self.workflow_id, HttpRequests(), use_caas=False)
+            self.assertEqual(mock_request.call_count, 3)
```

Comment:
This assert statement should be outside of the `with self.assertRaises` block so that the test can fail if it raises an error.

Reply:
good catch, thanks